### PR TITLE
Docs improvement and fixing bug with FID/IS metrics

### PR DESCRIPTION
**Pull request by gucifer**
Description:
Added explicitly that torchvision and scipy are needed for FID and InceptionScore.
Found a bug in InceptionScore due to pre-trained model returning logits instead of probabilities.
Fixed it by applying softmax function to deduce probabilities.
Check list:
- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/gan/inception_score.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ class InceptionScore(_BaseInceptionMetric):
 
     __ https://arxiv.org/pdf/1801.01973.pdf
 
+    .. note::
+        The default Inception model requires the `torchvision` module to be installed.
+
```
Comment (sdesrozis) [inline ignite/metrics/gan/inception_score.py:30]:
```suggestion
```

File: ignite/metrics/gan/fid.py [hunk 2]
```diff
@@ -71,6 +71,10 @@ class FID(_BaseInceptionMetric):
 
         __ https://github.com/mseitzer/pytorch-fid
 
+    .. note::
+        The default Inception model requires the `torchvision` module to be installed.
+        FID also requires `scipy` module for matrix square root calculations.
```
Comment (vfdev-5) [inline ignite/metrics/gan/fid.py:76]:
```suggestion
        FID also requires `scipy` library for matrix square root calculations.
```

File: ignite/metrics/gan/utils.py [hunk 3]
```diff
@@ -23,6 +23,7 @@ def __init__(self, return_features: bool, device: Union[str, torch.device] = "cp
             raise RuntimeError("This module requires torchvision to be installed.")
         super(InceptionModel, self).__init__()
         self._device = device
+        self.return_features = return_features
```
Comment (vfdev-5) [inline ignite/metrics/gan/utils.py:26]: Why do we need this as an attribute ?

Comment (gucifer) [reply ignite/metrics/gan/utils.py:26]: For the if else condition in the forward function

Comment (vfdev-5) [reply ignite/metrics/gan/utils.py:26]: We can append softmax layer once in the model definition.

Comment (gucifer) [reply ignite/metrics/gan/utils.py:26]: We can't just append to a sequential. We will have to create a new one.
If you are fine with that I will change the implementation that way.

**Review by sdesrozis — APPROVED** [review]
@gucifer  thank you ! LGTM !

**Comment by vfdev-5** [comment]
@sdesrozis I let you merge it if OK
